src/app/page.tsx

```diff
@@ -1,116 +1,121 @@
+    return { claimCondition };
+}
 
-	const currency = isERC1155Query.data
-		? claimCondition1155.data?.currency
-		: isERC721Query.data
-			? claimCondition721.data?.currency
-			: claimCondition20.data?.currency;
+async function getERC721Info() {
+    const [claimCondition, metadata] = await Promise.all([
+        getActiveClaimCondition721({ contract }),
+        getContractMetadata({ contract }),
+    ]);
+    const priceInWei = claimCondition?.pricePerToken;
+    const currency = claimCondition?.currency;
+    const currencyMetadata = currency
+        ? await getCurrencyMetadata({ contract: getContract({ address: currency, chain, client }) })
+        : undefined;
+    return {
+        displayName: metadata?.data?.name,
+        description: metadata?.data?.description,
+        pricePerToken: currencyMetadata && priceInWei ? Number(toTokens(priceInWei, currencyMetadata.decimals)) : null,
+        currencyMetadata,
+        metadata,
+        claimCondition,
+    };
+}
 
-	const currencyContract = getContract({
-		address: currency || "",
-		chain,
-		client,
-	});
+async function getERC1155Info() {
+    const [nft, claimCondition] = await Promise.all([
+        getNFT({ contract, tokenId }),
+        getActiveClaimCondition1155({ contract, tokenId }),
+    ]);
+    const priceInWei = claimCondition?.pricePerToken;
+    const currency = claimCondition?.currency;
+    const currencyMetadata = currency
+        ? await getCurrencyMetadata({ contract: getContract({ address: currency, chain, client }) })
+        : undefined;
+    return {
+        displayName: nft?.metadata.name,
+        description: nft?.metadata.description,
+        pricePerToken: currencyMetadata && priceInWei ? Number(toTokens(priceInWei, currencyMetadata.decimals)) : null,
+        currencyMetadata,
+        nft,
+        claimCondition,
+    };
+}
 
-	const currencyMetadata = useReadContract(getCurrencyMetadata, {
-		contract: currencyContract,
-		queryOptions: { enabled: !!currency },
-	});
+type ERCInfo = {
+    displayName?: string;
+    description?: string;
+    pricePerToken?: number | null;
+    currencyMetadata?: any;
+    metadata?: any;
+    claimCondition?: any;
+    nft?: any;
+};
 
-	const currencySymbol = currencyMetadata.data?.symbol || "";
+export default async function Home() {
+    const ercType = await getERCType();
+    let info: ERCInfo = {};
 
-	const pricePerToken =
-		currencyMetadata.data && priceInWei !== null && priceInWei !== undefined
-			? Number(toTokens(priceInWei, currencyMetadata.data.decimals))
-			: null;
+    switch (ercType) {
+        case "ERC20":
+            info = await getERC20Info();
+            console.log("ERC20", info);
+            break;
+        case "ERC721":
+            info = await getERC721Info();
+            console.log("ERC721", info
+            );
+            break;
+        case "ERC1155":
+            info = await getERC1155Info();
```

Comment:
what happens if any of these throw an error?

you should have some error handling here, should return a Error component with the error message